src/SDL_utils.c

```diff
@@ -123,7 +123,7 @@ static SDL_HashTable *SDL_objects;
 
 static Uint32 SDL_HashObject(const void *key, void *unused)
 {
-    return (Uint32)(uintptr_t)key;
+    return (Uint32)((uintptr_t)key & 0xffffffffu);
```

Comment:
Actually just to be _super_ annoying before we submit this upstream: Maybe we can use `SDL_UINT32_MAX` here just to make it clearer what this & is doing.